Userland/Libraries/LibWebView/WebContentClient.cpp

```diff
@@ -478,4 +478,14 @@ void WebContentClient::inspector_did_execute_console_script(String const& script
         m_view.on_inspector_executed_console_script(script);
 }
 
+Messages::WebContentClient::RequestWorkerAgentResponse WebContentClient::request_worker_agent()
+{
+    if (m_view.on_request_worker_agent) {
+        auto sockets = m_view.on_request_worker_agent();
```

Comment:
Does this have to be synchronous? We've been making an effort to make IPC all async, as sync has lead to deadlocks and crashes (when the browser process exits while WC is waiting).

Reply:
IMO we should allow sync IPC from WC -> UI, but not the other way around. Is there some reason we can't treat problems with that as normal bugs? :)

Reply:
That's probably fine, but then maybe we should be using the `send_sync_but_allow_failure` API here (TIL that method exists). Otherwise this will crash if the browser exits while this IPC is in flight.

Reply:
Using the send sync method now.